packages/blocks/src/text-input/index.js

```diff
@@ -20,6 +21,11 @@ const TextInput = ( { attributes, className } ) => {
 			if ( attributes.mandatory && isEmpty( value ) ) {
 				return __( 'This field is required', 'blocks' );
 			}
+			if ( attributes.validation && ! isEmpty( value ) ) {
+				for ( let i = 0; i < attributes.validation.length; i++ ) {
```

Comment:
Nit: Nowadays JS also offers `for ... of ...`, which would be a little shorter and assigns the array elements directly.